modules/ding_campaign_plus/plugins/content_types/ding_campaign_plus.inc

```diff
@@ -37,42 +38,25 @@ function ding_campaign_plus_relevant_content_type_render($subtype, $conf, $panel
       'raw' => arg(),
       'path' => drupal_get_path_alias($_GET['q']),
     ),
+    'search_request' => $currentResult ? $currentResult->getSearchRequest() : NULL,
   );
   $style = !empty($conf['style']) ? $conf['style'] : 'ribbon';
 
   // Check cache to see if campaign trigger has been calculated for this
   // context.
   $cid = _ding_campaign_plus_cache_id($context_data, $style);
   $cache = cache_get($cid);
+  $content = '';
   if (isset($cache->data) && REQUEST_TIME < $cache->expire) {
-    $campaign_nids = $cache->data;
+    $content = $cache->data;
   }
   else {
-    $campaign_nids = ding_campaign_plus_get_matches($context_data, $style);
-    cache_set($cid, $campaign_nids, 'cache', REQUEST_TIME + variable_get('ding_campaign_plus_expire', 900));
+    cache_set($cid . '-context', array($context_data, $style));
+    $content = '<div data-ding-campaign-plus-cid="' . $cid . '"></div>';
```

Comment:
Would be nice with a class here so it would be possible to have a spinner to indicate that more content is being loaded.